### PR TITLE
Follow up fix PR for S3C-1183/GCP-Tagging

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
Follow up fix for #1111
Includes:
Code cleanup, removed unnecessary async waterfall steps.
Adds more extensive tests for the put/get/del tagging tests.

### CODE REVIEW

**Comment by alexanderchan-scality** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_INTEGRATION_BRANCH=ft/GCP
SCALITY_FEDERATION_BRANCH=ft/S3C-1114/GCP-BACKEND
SCALITY_S3_BRANCH=fx/object-tagging

**Comment by alexanderchan-scality** [comment]
@ironman-machine try

**Comment by alexanderchan-scality** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_INTEGRATION_BRANCH=ft/S3C-1114/GCP-Backend
SCALITY_FEDERATION_BRANCH=ft/S3C-1114/GCP-BACKEND

**Comment by alexanderchan-scality** [comment]
@ironman-machine try

File: lib/data/external/GCP/GcpUtils.js [hunk 1]
```diff
@@ -118,35 +118,24 @@ function processTagSet(tagSet = []) {
...
-        const { Key: key, Value: value } = tag;
+    for (let i = 0; i < tagSet.length; ++i) {
+        const { Key: key, Value: value } = tagSet[i];
         if (key.length > 128) {
-            error = errors.InvalidTag
-                .customizeDescription(
-                    'The TagKey you have provided is invalid');
-            return false;
+            return errors.InvalidTag.customizeDescription(
+                'The TagKey you have provided is invalid');
```
Comment (jonathan-gramain) [inline lib/data/external/GCP/GcpUtils.js]: I suggest replacing `invalid` by `too long`, same below

Comment (jonathan-gramain) [reply lib/data/external/GCP/GcpUtils.js]: And you can provide the actual character count limit in the message as well.

**Comment by alexanderchan-scality** [comment]
Refactor is now part of #1141

**Pull request closed without merging**